app/models/notification.rb

```diff
@@ -79,7 +87,36 @@ def unarchive_if_updated
   def update_from_api_response(api_response, unarchive: false)
     attrs = Notification.attributes_from_api_response(api_response)
     self.attributes = attrs
+    update_subject
     unarchive_if_updated if unarchive
     save(touch: false) if changed?
   end
+
+  private
+
+  def download_subject
+    user.github_client.get(subject_url)
+  end
+
+  def update_subject
+    existing_subject = Subject.find_or_create_by(url: subject_url)
+
+    subject_updates = case subject_type
+    when 'Issue', 'PullRequest'
+      remote_subject = download_subject
+       {
+        state: remote_subject.merged_at.present? ? 'merged' : remote_subject.state,
+        author: remote_subject.user.login
+      }
+    when 'Commit', 'Release'
+      remote_subject = download_subject
+      subject_updates = {
+        author: remote_subject.author.login
+      }
+    else
+      {}
+    end
```

Comment:
Although I wish this were somehow a part of DownloadService, I do like this much better than my somewhat overengineered STI thing I have going. Simpler.